library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -366,6 +367,9 @@ public int getScanIntervalMillis()
    */
   public void setScanIntervalMillis(int scanIntervalMillis)
   {
+    if (scanIntervalMillis < 0) {
+      throw new IllegalArgumentException("scanIntervalMillis should be greater than or equal to 0.");
+    }
```

Comment:
Can we add the validation annotation to scanIntervalMillis, instead fixing it in setter method ?